server/src/operators/chunk_operator.rs

```diff
@@ -1346,6 +1346,7 @@ pub enum HighlightStrategy {
     V1,
 }
 
+#[inline(never)]
 pub fn get_highlights_with_exact_match(
```

Comment:
This is where I suspect the fishy code is at, but added traces at other spots just in case